vm/src/stdlib/random.rs

```diff
@@ -1,15 +1,82 @@
 //! Random module.
 
+use std::cell::RefCell;
+
+use num_bigint::{BigInt, Sign};
+
 use rand::distributions::Distribution;
+use rand::{RngCore, SeedableRng};
+use rand::rngs::SmallRng;
 use rand_distr::Normal;
 
-use crate::pyobject::{PyObjectRef, PyResult};
+use crate::function::OptionalArg;
+use crate::obj::objtype::PyClassRef;
+use crate::pyobject::{PyClassImpl, PyObjectRef, PyRef, PyValue, PyResult};
+
 use crate::vm::VirtualMachine;
 
+#[pyclass(name = "Random")]
+#[derive(Debug)]
+struct PyRandom {
+    rng: RefCell<SmallRng>
+}
+
+impl PyValue for PyRandom {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("_random", "Random")
+    }
+}
+
+#[pyimpl]
+impl PyRandom {
+    #[pyslot(new)]
+    fn new(cls: PyClassRef, vm: &VirtualMachine) -> PyResult<PyRef<Self>> {
+        PyRandom {
+            rng: RefCell::new(SmallRng::from_entropy())
+        }.into_ref_with_type(vm, cls)
+    }
+
+    #[pymethod] 
+    fn seed(&self, n: Option<usize>, vm: &VirtualMachine) -> PyResult {
+        let rng = match n {
+            None => SmallRng::from_entropy(),
+            Some(n) => {
+                let seed = n as u64;
+                SmallRng::seed_from_u64(seed)
```

Comment:
It would be nice to support larger seeds (via bigints), but this requires more work and could be done later (likely in a backwards incompatible way).